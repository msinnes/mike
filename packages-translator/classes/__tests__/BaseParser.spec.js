const BaseParser = require('../BaseParser');

describe('BaseParser', () => {
  it('should be an abstract class', () => {
    expect(() => {
      BaseParser('string');
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      new BaseParser('string');
    }).toThrowErrorMatchingSnapshot();
  });
});