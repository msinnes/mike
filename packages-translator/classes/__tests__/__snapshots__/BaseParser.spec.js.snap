// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseParser should be an abstract class 1`] = `"ClassError: Cannot call a class as a function"`;

exports[`BaseParser should be an abstract class 2`] = `"ClassError: Abstract classes cannot be instantiated"`;
